src/main/java/ch/njol/skript/lang/parser/ParserInstance.java

```diff
@@ -89,20 +93,79 @@ public Node getNode() {
 	}
 	
 	@Nullable
-	public Config getCurrentScript() {
+	public Script getCurrentScript() {
 		return currentScript;
 	}
-	
+
+	// Just for the deprecated method so that we don't bother creating empty maps
+	private static final HashMap<String, String> DUMMY_MAP = new HashMap<>(0);
+
+	/**
+	 * Deprecated. Use {@link ch.njol.skript.structures.StructOptions#getOptions(Script)} instead.
+	 */
+	@Deprecated
 	public HashMap<String, String> getCurrentOptions() {
-		return currentOptions;
+		Script script = getCurrentScript();
+		if (script == null)
+			return DUMMY_MAP;
+		HashMap<String, String> options = StructOptions.getOptions(script);
+		if (options == null)
+			return DUMMY_MAP;
```

Comment:
what happens if the consumer of this map modifies it? now future callers no longer get an empty map